src/libraries/System.Security.Cryptography.X509Certificates/tests/DynamicChainTests.cs

```diff
@@ -201,6 +201,97 @@ void CheckChain()
+                        certificateAuthority: true,
+                        hasPathLengthConstraint: true,
+                        pathLengthConstraint: 2,
+                        critical: true));
+
+                CertificateRequest noSigningIntermediateReq = new CertificateRequest(
+                    "CN=Intermediate",
+                    key,
+                    HashAlgorithmName.SHA256,
+                    RSASignaturePadding.Pkcs1);
+
+                noSigningIntermediateReq.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(
+                        certificateAuthority: false,
+                        hasPathLengthConstraint: true,
+                        pathLengthConstraint: 0,
+                        critical: true));
+
+                CertificateRequest leafReq = new CertificateRequest(
+                    "CN=Leaf",
+                    key,
+                    HashAlgorithmName.SHA256,
+                    RSASignaturePadding.Pkcs1);
+
+                DateTimeOffset notBefore = DateTimeOffset.UtcNow.AddDays(-1);
+                DateTimeOffset notAfter = notBefore.AddDays(30);
+
+                using (X509Certificate2 root = rootReq.CreateSelfSigned(notBefore, notAfter))
+                using (X509Certificate2 signingIntermediate = signingIntermediateReq.Create(root, notBefore, notAfter, serialNumber))
+                using (X509Certificate2 noSigningIntermediate = noSigningIntermediateReq.Create(root, notBefore, notAfter, serialNumber))
+                using (X509Certificate2 signingIntermediateWithKey = signingIntermediate.CopyWithPrivateKey(key))
+                using (X509Certificate2 leaf = leafReq.Create(signingIntermediateWithKey, notBefore, notAfter, serialNumber))
+                {
+                    X509Chain chain = new X509Chain();
+                    chain.ChainPolicy.ExtraStore.Add(root);
+                    chain.ChainPolicy.ExtraStore.Add(noSigningIntermediate);
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                    Assert.False(chain.Build(leaf));
+                    
+                    bool verifiedIntermediate = false;
+
+                    foreach (X509ChainElement element in chain.ChainElements)
+                    {
+                        if (element.Certificate.Subject == "CN=Intermediate")
+                        {
+                            const X509ChainStatusFlags expectedFlag = X509ChainStatusFlags.InvalidBasicConstraints;
+                            X509ChainStatusFlags intermediateStatus = element.AllStatusFlags();
+                            Assert.True((intermediateStatus & expectedFlag) == expectedFlag, $"Has {expectedFlag} flag");
+                            verifiedIntermediate = true;
+                            break;
+                        }
+                    }
+
+                    Assert.True(verifiedIntermediate, "verify leaf certificate");
+                }
```

Comment:
Do we already have a test for pathLen exceeded?  I feel like the easy overload on CertificateRequest will check that and block it, but if you use the X509SignatureGenerator version (where it can't see the parent cert) then it obviously won't.

Reply:
(Really the question here is is there an extra extra error that appears for pathlen exceeded)

Reply:
> Really the question here is is there an extra extra error that appears for pathlen exceeded

Yeah, It's reported as BasicConstraintsPathLen which this test handled both since Apple reports both in this case (It's a CA:FALSE and has a path length violation) - I will split this in to two separate tests.

> X509SignatureGenerator

Oh that's much better than the weird intermediate swap-out I did. I completely missed `Create`.